docs/google/devices.md

```diff
@@ -0,0 +1,194 @@
+
+Using built-in optimizers will allow you to compile to the correct .  As they are
+automated solutions, they will not always perform as well as a hand-crafted solution, but
+it provides a good starting point for creating a circuit that is likely to run successfully
+on hardware.
+
+```python
+import cirq
+import cirq.google as cg
+
+
+# Create your circuit here
+my_circuit = cirq.Circuit()
+
+# Convert the circuit onto a Google device.
+# Specifying a device will verify that the circuit satisifies constraints of the device
+# Specifying an optimizer type (e.g. 'sqrt_iswap' or 'sycamore') will convert to the appropriate
+# gate set and also do basic optimization, such as combining successive one-qubit gates. 
+sycamore_circuit = cg.optimized_for_sycamore(my_circuit, new_device=cg.Sycamore, optimizer_type='sqrt_iswap')
+```
+
+### Good moment structure
+
+In an attempt to fulfill the user's desire, quantum devices will respect the moment
+structure of circuits to the greatest extent possible.
+
+To this end, it is important that the moment structure of the circuit is kept as
+short and concise as possible.  The length of a moment will generally be the length
+of the longest gate in the moment, so keeping gates with similar durations together
+will generally benefit the performance.
+
+In particular, keep measurement gates in the same moment and make sure that any
+circuit optimizers run do not alter this by pushing measurements forward.  This
+behavior can be avoided by measuring all qubits with a single gate or by adding
+the measurement gate after all optimizers have run.
+
+### Short gate depth
+
+In the current NISQ (noisy intermediate scale quantum) era, gates and devices still
+have significant error.  Both gate errors and T1 decay rate can cause long circuits
+to have noise that overwhelms any signal in the circuit.  Also, circuits that are
+much too long can even be rejected due to various limits in the system.
+
+The recommended gate depths vary significantly with the structure of the circuit itself
+and will likely increase as the devices improve.  However, circuits with more than
+50 moments will generally struggle to retain any information.
+
+### Keep qubits busy
+ 
+Qubits that remain idle for long periods tend to dephase and decoher.  Inserting a
+[Spin Echo](https://en.wikipedia.org/wiki/Spin_echo) into your circuit, such as a pair
+of involutions, such as two successive Pauli Y gates, will generally increase
+performance of the circuit.
+
+### Alternate single-qubit and two-qubit layers
+
+Devices are generally calibrated to circuits that alternate single-qubit gates with
+two-qubit gates in each layer.  Staying close to this paradigm will often improve
+performance of circuits.
+
+Devices generally operate in the Z basis, so that rotations around the Z axis will become
+book-keeping measures rather than physical operations on the device.  These operations
+should be aggregated into their own moment, if possible.
```

Comment:
open or link to github issue requesting better cirq features to do this

Reply:
We have an internal bug for this, but no cirq issue.  I am going to punt on creating a GH issue for this until we decide what we want.  I want to keep general advice separate from creating automated functionality to do these optimizations.